src/System.CommandLine/RootCommand.cs

```diff
@@ -2,13 +2,18 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.IO;
+using System.Linq;
 using System.Reflection;
 
 namespace System.CommandLine
 {
     public class RootCommand : Command
     {
-        public RootCommand(string description = "") : base(ExeName, description)
+        public RootCommand(string description = "") : this(ExeName, description)
+        {
+        }
+
+        public RootCommand(string exeName, string description = "") : base(exeName, description)
```

Comment:
For commands, options, and arguments, the constructor that accepts a string is used to set the `Name` property, so for consistency we may want to deprecate the existing constructor here.

What do we lose by not adding this constructor?

Reply:
This constructor is optional. The alternative is using the first command line invocation argument, which (I believe) may not be the executable (e.g. using a test runner). If that's _not_ fine for someone, the constructor offered an alternative. I'm fine with not adding it, otherwise being able to _set_ the ExeName property (make it a public settable member) would work too.

Reply:
You can set `RootCommand.Name` which is the equivalent of passing it through the constructor.

Reply:
Ok so I have a local commit ready, but I want to make sure I understood your feedback correctly.

Right now, every class that inherits Symbol have a name/alias which can be (sometimes optionally) specified in the constructor. The first argument being the name except for the root command, which has a description as it's first constructor argument.

So if I understood correctly, you would prefer deprecating the _current_ constructor which receives the description, or the one I proposed in my PR? In any case I can simply remove the new constructor from this PR, since the alternative works for me anyway, though it is not as "reliable" as a specified name.

Also, setting `.Name` would actually _add_ an alias, due to the current implementation. I think this would work anyway since only the longuest/latest alias is used, but still has low-impact side-effects.

So, the easiest option is to just remove the new constructor to merge the part that works, if that's what you want to go with, just let me know! (I don't want to just update the PR before sharing this info, since I wanted this feedback to be visible inside the same conversation)

Reply:
Yes, this is correct. My suggestion is to leave out the new constructor. We'll likely remove the other constructor later. So far the only user scenario I've seen for setting the `RootCommand.Name` has been for help output.

The point about it adding an alias when setting `Name` is a good one. This looks like a possible bug. The intent is that the actual executable name should always be recognized by the parser and shouldn't be overwritable, since e.g. for suggestions we have to parse the complete command line which might include the executable name in various forms (with or without path, with or without extension).